spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/config/KinesisBinderConfiguration.java

```diff
@@ -0,0 +1,47 @@
+package org.springframework.cloud.stream.binder.kinesis.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.binder.kinesis.KinesisMessageChannelBinder;
+import org.springframework.cloud.stream.binder.kinesis.properties.KinesisBinderConfigurationProperties;
+import org.springframework.cloud.stream.binder.kinesis.properties.KinesisExtendedBindingProperties;
+import org.springframework.cloud.stream.binder.kinesis.provisioning.KinesisStreamProvisioner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.codec.Codec;
+
+/**
+ * 
+ * @author Peter Oates
+ *
+ */
+//@Configuration
+//@ConditionalOnMissingBean(Binder.class)
+//@Import({KryoCodecAutoConfiguration.class, PropertyPlaceholderAutoConfiguration.class})
+@EnableConfigurationProperties({KinesisBinderConfigurationProperties.class, KinesisExtendedBindingProperties.class})
+public class KinesisBinderConfiguration {
+
+	@Autowired
+	private Codec codec;
+
+	@Autowired
+	private KinesisBinderConfigurationProperties configurationProperties;
+
+
+	@Bean
+	KinesisStreamProvisioner provisioningProvider() {
+		return new KinesisStreamProvisioner(this.configurationProperties);
+	}
+
+	@Bean
+	KinesisMessageChannelBinder kinesisMessageChannelBinder() {
+		
+		KinesisMessageChannelBinder kinesisMessageChannelBinder = new KinesisMessageChannelBinder(
+				this.configurationProperties, provisioningProvider());
```

Comment:
The `AWSCredentialsProvider` and `Region` must be injected from here into the `KinesisMessageChannelBinder` with appropriate delegation to the `AmazonKinesisAsyncClientBuilder`.

You can just `@Autowired AWSCredentialsProvider` and `@Autowired RegionProvider` into this `KinesisBinderConfiguration`.

The particular instances of those beans must be supplied by the auto-configuration from the SC-AWS dependency.